### PR TITLE
stm32h7: Use enums for the prescalers in the RCC PLL config struct

### PR DESCRIPTION
**Pull request by donotnoot**
This had me confused for a little while until I realised that the fields don't take an integer value representing the clock dividers but actually a value that goes into the relevant registers.

It would have saved me a little bit of time if the documentation was a bit more explicit, so hoping this makes it a bit clearer for the next user :)

### CODE REVIEW

**Comment by donotnoot** [comment]
Ah, just saw https://github.com/libopencm3/libopencm3/issues/1299#issuecomment-773631832. I can give that a shot too? I do think enums would be even better.

**Comment by vielster** [comment]
Assuming updating to enums doesn't break any backward compatibility (i.e. you don't have to rename anything) then I'm all for it.

**Comment by donotnoot** [comment]
Cool, that's it with enums and backwards compatible (if we ignore the fact that the struct is not binary compatible, which I guess is fine...). I generated the Doxygen docs and they look pretty good too. Everything is dead obvious now.

I'm not too sure about the flash waitstates, we could do the same? But that's a slightly bigger change and honestly not a big deal since a plain integer in that field works.

**Comment by karlp** [comment]
what, no, this looks horrible!  now youv'e got splatter of everything!  I'm not ok with duplicating it all, no thanks.  you can make a case that this is new enough that you can replace the defines with enums, and you can easily add doxygen links to the existing groupings that might be enough without adding enums, but I'm not having _both_ the defines _and_ enums!

**Comment by donotnoot** [comment]
Alright, that's what you meant! Now I know. See the update :+1:

I checked the generated documentation and it looks alright. I'm not familiar enough with Doxygen to know if that's correct/acceptable, please do let me know.

File: include/libopencm3/stm32/h7/rcc.h [hunk 1]
```diff
@@ -182,29 +182,37 @@ LGPL License Terms @ref lgpl_license
...
-#define RCC_D1CFGR_D1HPRE_BYP       0x0
-#define RCC_D1CFGR_D1HPRE_DIV2      0x8
-#define RCC_D1CFGR_D1HPRE_DIV4      0x9
-#define RCC_D1CFGR_D1HPRE_DIV8      0xA
-#define RCC_D1CFGR_D1HPRE_DIV16     0xB
-#define RCC_D1CFGR_D1HPRE_DIV64     0xC
-#define RCC_D1CFGR_D1HPRE_DIV128    0xD
-#define RCC_D1CFGR_D1HPRE_DIV256    0xE
-#define RCC_D1CFGR_D1HPRE_DIV512    0xF
+enum rcc_core_pre {
```
Comment (karlp) [inline include/libopencm3/stm32/h7/rcc.h:185]: I think the enum naming should stick with the existing naming convention, something like rcc_d1cfgr_[your-free-text-here]  things like "rcc_core_" are going to start colliding or being non-visible quite quickly if they're not standardized.

File: include/libopencm3/stm32/h7/rcc.h [hunk 2]
```diff
@@ -455,13 +468,13 @@ struct rcc_pll_config {
...
-  uint8_t ppre3;                    /**< APB3 Peripheral prescaler note: domain 1. */
-  uint8_t ppre4;                    /**< APB4 Peripheral prescaler note: domain 3. */
-  uint8_t flash_waitstates;         /**< Latency Value to set for flahs. */
+  enum rcc_core_pre core_pre;       /**< Core prescaler. */
+  enum rcc_domain1_hpre hpre;       /**< HCLK3 prescaler. */
+  enum rcc_domain2_ppre ppre1;      /**< APB1 Peripheral prescaler. */
+  enum rcc_domain2_ppre ppre2;      /**< APB2 Peripheral prescaler. */
+  enum rcc_domain1_ppre ppre3;      /**< APB3 Peripheral prescaler. */
+  enum rcc_domain3_ppre ppre4;      /**< APB4 Peripheral prescaler. */
+  uint8_t flash_waitstates;         /**< Latency Value to set for flash. FLASH_ACR_LATENCY_xWS value. */
```
Comment (karlp) [inline include/libopencm3/stm32/h7/rcc.h:477]: this is out of scope, and aren't the flash wait states pure numeric anyway?

Comment (vielster) [reply include/libopencm3/stm32/h7/rcc.h:477]: These shift value for this macro is zero, so it works either way, but I agree it's sorta outside of the scope of changing the rcc regs to enums

**Comment by karlp** [comment]
still needs the agreed enum name changes.

**Comment by donotnoot** [comment]
Closing due to lack of time/energy.
